Show shared access policy in volume details

DarkAEther opened the pull request:
Adds Shared Access Policy to volume details view.
Closes #4402

deviantony commented:
/azp run

deviantony reviewed (changes requested):
Super sorry for the late review @DarkAEther !

Overall this change is looking good but I think it would be even better if we added a tooltip next to the value of the Shared Access Policy with the tooltip message being set to the more information message we have in the cluster setup.

![Portainer-172-17-0-5-30778](https://user-images.githubusercontent.com/5485061/100572421-4abdbb00-333a-11eb-8eeb-8ae598db3870.png)

![Portainer-172-17-0-5-30778 (1)](https://user-images.githubusercontent.com/5485061/100572446-5a3d0400-333a-11eb-9b08-2927bc86d49f.png)

DarkAEther commented:
> Super sorry for the late review @DarkAEther !
> 
> Overall this change is looking good but I think it would be even better if we added a tooltip next to the value of the Shared Access Policy with the tooltip message being set to the more information message we have in the cluster setup.

Hey @deviantony! No worries! Thank you for taking the time to review the PR. I've added in the tooltip as requested.

deviantony reviewed (changes requested):
This PR includes too many files/commit, I think you'll need to rebase it properly or to re-create it.

DarkAEther commented:
> This PR includes too many files/commit, I think you'll need to rebase it properly or to re-create it.

I'd like to sincerely apologize for this. I've made changes to fix it. I hope it's alright now.

Thank you!

deviantony commented:
@pull-dog down

deviantony commented:
/azp run

deviantony commented:
@pull-dog down

deviantony reviewed (changes requested):
Overall looking good!

One small change required:

- [ ] When the storage option supports both RWO and RWX, we should display RWX in the volume details (as RWX also means it supports RWO).

![Portainer-172-17-0-8](https://user-images.githubusercontent.com/5485061/100795820-8a90b980-3484-11eb-9e51-6d150952f5c3.png)

![Portainer-172-17-0-8 (1)](https://user-images.githubusercontent.com/5485061/100795830-8d8baa00-3484-11eb-9013-011cb564934a.png)

DarkAEther commented:
> Overall looking good!
> 
> One small change required:
> 
> * [ ]  When the storage option supports both RWO and RWX, we should display RWX in the volume details (as RWX also means it supports RWO).
> 
> ![Portainer-172-17-0-8](https://user-images.githubusercontent.com/5485061/100795820-8a90b980-3484-11eb-9e51-6d150952f5c3.png)
> 
> ![Portainer-172-17-0-8 (1)](https://user-images.githubusercontent.com/5485061/100795830-8d8baa00-3484-11eb-9013-011cb564934a.png)

Hi @deviantony! Thank you for pointing that out. I've made changes to accommodate that. However, I'm a little doubtful regarding the way I have implemented this. My implementation depends on the fact that either:
1. There is only one access mode allowed at a time.
2. If multiple access modes are selected (RWO+RWX), then RWX appears last in the `this.volume.PersistentVolumeClaim.StorageClass.AccessModes` array.

deviantony commented:
/azp run

deviantony commented:
@pull-dog up

deviantony commented:
@pull-dog down

deviantony reviewed (approved):
LGTM

itsconquest commented:
@DarkAEther it appears there is a merge conflict that needs to be resolved. Also I can see in the commit history that this PR has a bunch of other commit history in it. Are you able to try and resolve this or open a new PR with just your changes? 
Once that is done, I can test and merge your changes 👍🏻

DarkAEther commented:
> @DarkAEther it appears there is a merge conflict that needs to be resolved. Also I can see in the commit history that this PR has a bunch of other commit history in it. Are you able to try and resolve this or open a new PR with just your changes?
> Once that is done, I can test and merge your changes 👍🏻

Thanks for having a look @itsconquest ! I think the easiest way to clean this up would be a new PR, so I opened up #4707 and will close this PR.
